dependency_support/load_external.bzl

```diff
@@ -286,3 +286,15 @@ def load_external_repositories():
             "https://github.com/nlohmann/json/archive/refs/tags/v3.10.2.tar.gz",
         ],
     )
+
+    # Version 1.4.7 released on 17.12.2020
+    # https://github.com/facebook/zstd/releases/tag/v1.4.7
+    # Updated 23.11.2023
+    http_archive(
+        name = "com_github_facebook_zstd",
+        sha256 = "192cbb1274a9672cbcceaf47b5c4e9e59691ca60a357f1d4a8b2dfa2c365d757",
+        strip_prefix = "zstd-1.4.7",
+        urls = ["https://github.com/facebook/zstd/releases/download/v1.4.7/zstd-1.4.7.tar.gz"],
+        build_file = "@//dependency_support/com_github_facebook_zstd:bundled.BUILD.bazel",
+        patches = ["@com_google_xls//dependency_support/com_github_facebook_zstd:decodecorpus.patch"],
```

Comment:
can you add a comment about the patch? has it been submitted upstream?

Reply:
It was not submitted upstream yet. The patch modifies `decodecorpus` utility from zstd library. Unmodified tool allows generating valid zstd frames with randomized contents and size. With our modifications it is possibile to generate only some parts of the zstd frame. In this case we modify `decodecorpus` so that it allows generating only frame headers.